libs/cart/src/effects/cart-item.effects.ts

```diff
@@ -65,29 +74,45 @@ export class DaffCartItemEffects<
   add$ = this.actions$.pipe(
     ofType(DaffCartItemActionTypes.CartItemAddAction),
     switchMap((action: DaffCartItemAdd<U>) =>
-      this.driver.add(
-        this.storage.getCartId(),
-        action.input
-      ).pipe(
-        map((resp: V) => new DaffCartItemAddSuccess(resp)),
+      combineLatest([
+				this.store.pipe(
+					select(getDaffCartItemEntitiesSelectors<V, X, T>().selectAllCartItems),
+					take(1)
+				),
+				this.driver.add(this.storage.getCartId(), action.input)
+			]).pipe(
+        map(([cartItems, resp]: [T[], V]) => new DaffCartItemAddSuccess({
+					...resp,
+					items: this.updateAddedCartItemState(cartItems, <T[]>resp.items)
+				})),
         catchError(error => of(new DaffCartItemAddFailure('Failed to add cart item')))
       )
     )
-  )
+	)
 
   @Effect()
   update$ = this.actions$.pipe(
     ofType(DaffCartItemActionTypes.CartItemUpdateAction),
     switchMap((action: DaffCartItemUpdate<T>) =>
-      this.driver.update(
-        this.storage.getCartId(),
-        action.itemId,
-        action.changes
-      ).pipe(
-        map((resp: V) => new DaffCartItemUpdateSuccess(resp)),
-        catchError(error => of(new DaffCartItemUpdateFailure('Failed to update cart item')))
-      )
-    )
+			this.driver.update(
+				this.storage.getCartId(),
+				action.itemId,
+				action.changes
+			).pipe(
+				map((resp: V) => new DaffCartItemUpdateSuccess({
+					...resp,
+					items: this.updateMutatedCartItemState(<T[]>resp.items, action.itemId)
+				})),
+				catchError(error => of(new DaffCartItemUpdateFailure('Failed to update cart item')))
+			)
+		)
+	)
+
+	@Effect()
+  resetCartItemStateAfterChange$ = this.actions$.pipe(
+		ofType(DaffCartItemActionTypes.CartItemAddSuccessAction, DaffCartItemActionTypes.CartItemUpdateSuccessAction),
+		debounceTime(this.cartItemStateDebounceTime),
+		map(() => new DaffCartItemStateReset())
```

Comment:
I would think that the reset would need to know the item ID. Maybe a global reset is the desired behavior here, I'm not sure.

Reply:
Yea, I explain in the PR description why I decided to go with a global reset instead of tracking the state changes individually.